adagrams/game.py

```diff
@@ -1,11 +1,87 @@
+LETTER_POOL = {
+    'A': 9, 'B': 2, 'C': 2,'D': 4, 'E': 12, 'F': 2, 'G': 3, 'H': 2, 'I': 9, 
+    'J': 1, 'K': 1, 'L': 4, 'M': 2, 'N': 6, 'O': 8, 'P': 2,  'Q': 1, 'R': 6, 
+    'S': 4, 'T': 6, 'U': 4, 'V': 2, 'W': 2, 'X': 1, 'Y': 2, 'Z': 1
+}
+
 def draw_letters():
-    pass
+    letters_not_drawn = list(LETTER_POOL.keys())
+    letters_drawn = []
+    letter_count = 0
 
-def uses_available_letters(word, letter_bank):
-    pass
+    for letter in letters_not_drawn:
+        if letter not in letters_drawn and len(letters_drawn) < 10:
+            letters_drawn.append(letter)
+            letter_count += 1
+    return letters_drawn
 
+def uses_available_letters(word, letter_bank):
+    word = word.upper()
+    word_dict = {}
+    is_valid = False
+    letter_count = 0
+    for letter in word:
+        if letter in letter_bank and word.count(letter) <= letter_bank.count(letter):
+            word_dict[letter] = True
+            letter_count += 1
+            is_valid = True
+        elif not letter in letter_bank:
+            word_dict[letter] = False
+            is_valid = False
+    return is_valid
```

Comment:
I think we can get rid of the `word_dict` and `letter_count` altogether! Both are created and updated, but then they aren't used anywhere to calculate our final result.

Secondly, there's a logic error that the tests didn't pick up on. Let's take the word "ALABAMA" for example. Based on this function:
```python
letter_bank = ["F","A","B","D","A","C","C","L","M","N"]
uses_available_letters("ALABAMA", letter_bank)
```
it returns `True`, which it shouldn't. What's causing this? Well, let's take "A" through your conditionals.

`if letter in letter_bank and word.count(letter) <= letter_bank.count(letter)` 4<= 2 so, no this doesn't execute.

`elif not letter in letter_bank` "A" is in the letter_bank, so this doesn't execute either.

What can we add to this function to fix this?